frontend/src/pages/Login.js

```diff
@@ -0,0 +1,152 @@
+  const [password, setPassword] = useState("")
+  const [mode, setMode] = useState("signin")
+
+  const accessToken = useSelector((store) => store.user.accessToken)
+  const error = useSelector((store) => store.user.error)
+  console.log('Error: ', error)
+
+  const dispatch = useDispatch()
+  const navigate = useNavigate()
+
+  useEffect(() => {
+    if (accessToken) {
+      navigate("/")
+    }
+  }, [accessToken, navigate])
+
+  const onFormSubmit = (event) => {
+    event.preventDefault()
+
+    const options = {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({ username, password }),
+    }
+
+    fetch(API_URL(mode), options)
+      .then((res) => res.json())
+      .then((data) => {
+        if (data.success) {
+          batch(() => {
+            dispatch(user.actions.setUserId(data.response.userId))
+            dispatch(user.actions.setUsername(data.response.username))
+            dispatch(user.actions.setAccessToken(data.response.accessToken))
+            dispatch(settings.actions.setWorkMinutes(data.response.workMinutes))
+            dispatch(settings.actions.setShortBreakMinutes(data.response.shortBreakMinutes))
+            dispatch(settings.actions.setLongBreakMinutes(data.response.longBreakMinutes))
+            dispatch(user.actions.setError(null))
+          })
+        } else {
+          batch(() => {
+            dispatch(user.actions.setUserId(null))
+            dispatch(user.actions.setUsername(null))
+            dispatch(user.actions.setAccessToken(null))
+            dispatch(settings.actions.setWorkMinutes(25))
+            dispatch(settings.actions.setShortBreakMinutes(5))
+            dispatch(settings.actions.setLongBreakMinutes(15))
+            dispatch(user.actions.setError(data.response))
+          })
+        }
+      })
+  }
+
+  return (
+    <>
+      <LoginContainer>
+        <FormWrapper>
+          <h3 style={{ color: "var(--red)" }}>
+            {mode === "signin"
+              ? "Log in to your account"
+              : "Create new account"}
+          </h3>
+          <form onSubmit={onFormSubmit}>
+            <UserInfoWrapper>
+              <legend>Username:</legend>
+              <UserInput
+                type="text"
+                value={username}
+                onChange={(e) => setUsername(e.target.value)}
+              />
+            </UserInfoWrapper>
+            <UserInfoWrapper>
+              <legend>Password:</legend>
+              <UserInput
+                type="password"
+                value={password}
+                onChange={(e) => setPassword(e.target.value)}
+              />
+            </UserInfoWrapper>
+            {error && (
+              <div>
+                <p
+                  style={{
+                    color: "red",
+                    textAlign: "center",
+                    fontWeight: "bold",
+                  }}
+                >
```

Comment:
This element could be styled component, don't you think?

Reply:
Absolutely